test/e2e/app-dir/third-parties/app/gtm/page.js

```diff
@@ -0,0 +1,27 @@
+'use client'
+
+import { useEffect } from 'react'
+import { useGoogleTagManager } from '@next/third-parties/google'
+
+const Page = () => {
+  const { init, sendData } = useGoogleTagManager()
+
+  useEffect(() => {
+    init({ id: 'GTM-XYZ' })
+  }, [init])
+
+  const onClick = () => {
+    sendData({ event: 'buttonClicked', value: 'xyz' })
+  }
+
+  return (
+    <div class="container">
+      <h1>GTM</h1>
+      <button id="gtm-send" onClick={onClick}>
+        Click
+      </button>
+    </div>
+  )
+}
+
+export default Page
```

Comment:
We can do this in a separate PR, but we should update the GTM example in the `examples/` repo to show this newer approach

Reply:
Is the double `<GoogleTagManager gtmId="GTM-XYZ" />`  on line 13 and 18 an oversight? Same in `test/e2e/third-parties/pages/gtm.js`

Reply:
I had thought the same in the beginning, but I recall @janicklas-ralph mentioning that it is was deliberate to test that the behavior if an identical GTM containers is included by accident

@janicklas-ralph Can you clarify, and if so, add a note here in a future PR? (not urgent)